### PR TITLE
Docs/react media store

### PR DESCRIPTION
**Pull request by cjpillsbury**
* Update docs to call out react-specific usage. Add extensive docs for react hooks
* (semi-hacky) refactor sandpack componentization in astro docs to be able to use for e.g. react templates.
* Improve react hooks material ui nextjs example. Default to use mux-video for the media state owner/media ref.
* Improve types and constants for react hooks usage.

* Material UI Hooks usage example (updated to use `<mux-video>` for state demo purposes): https://media-chrome-demo-nextjs-git-fork-cjpillsbury-docs-r-1582c8-mux.vercel.app/material-ui-player-chrome
* React Docs Pages
  * Get Started (updated to include Sandpack example) - https://media-chrome-docs-git-fork-cjpillsbury-docs-react-me-657343-mux.vercel.app/docs/en/react/get-started
  * Media Store Hooks - https://media-chrome-docs-git-fork-cjpillsbury-docs-react-me-657343-mux.vercel.app/docs/en/react/get-started

### CODE REVIEW

File: src/js/utils/time.js [hunk 1]
```diff
@@ -91,13 +91,16 @@ export function formatTime(seconds, guide) {
   }
 
   // Check if we need to show hours
+  // @ts-ignore
```
Comment (cjpillsbury) [inline src/js/utils/time.js:94]: added for quick fix for TS validity

File: docs/src/components/SandpackContainerReact.astro [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+  previewAspectRatio = 16 / 9,
+  stacked,
+  reversed,
+  ...props
+} = Astro.props as Props;
+
+if (typeof editorHeight === 'number') editorHeight = `${editorHeight}px`;
+---
+<!--
+  @TODO This is basically a duplicate of SandpackContainer.astro but uses SandpackBase instead of the more opinionated Sandpack React component.
```
Comment (cjpillsbury) [inline docs/src/components/SandpackContainerReact.astro:16]: NOTE: This todo is a callout on the refactor to support sandpack templates like `'vite-react-ts'` for react cases without a full rearchitecture to also handle convenience crud for standard web component use cases/backwards compatibility. Didn't want to blow up effort here, but if someone else sees a quick+clean alternative solution, happy to accept contribs.

**Comment by luwes** [comment]
not sure if it's an easy fix but could the 2nd column be made more wide, in a few tables they look squished

SCR-20240411-kesi

**Comment by cjpillsbury** [comment]
> not sure if it's an easy fix but could the 2nd column be made more wide, in a few tables they look squished
> 

Yeah I didn't see any way to resolve that quickly, but if you find something, keep me posted.

**Review by luwes — APPROVED** [review]
LGTM! didn't go through with a fine comb but feel confident to approve

**Pull request merged**